File: nodes/number/number.process.py
async def process(inputs, settings, config, nodeConfig):
    """
    Process function for the Number node.
    Outputs a constant numeric value, with optional input override.
    """
    # Use input value if provided, otherwise use the setting value
    if "value" in inputs and inputs["value"] is not None:
        value = float(inputs["value"])
    else:
        value = float(settings.get("value", 0))
    
    # Ensure the value is within min and max bounds
    min_val = float(settings.get("min", 0))
    max_val = float(settings.get("max", 100))
    
    value = max(min_val, min(max_val, value))

    return {"value": value} 

File: nodes/number/number.process.js
export default async ({inputs, settings, config}) => {

  
  // Use input value if provided, otherwise use the setting value
  let value = (inputs.value !== undefined && inputs.value !== null) ? Number(inputs.value) : Number(settings.value);
  
  // Ensure the value is within min and max bounds
  const min = Number(settings.min);
  const max = Number(settings.max);
  
  if (!isNaN(min) && !isNaN(max)) {
    value = Math.max(min, Math.min(max, value));
  }

  return { value };
}; 